### PR TITLE
[MRG+2] Plot errorbars when fmt == 'none'

### LINKED ISSUE #8117: Trying to plot only error bars (with no line connecting data points) results in empty plot
mpl2.0, regression from 1.5.3 and thus marked as release critical, but feel free to relabel/remilestone.
```
plt.errorbar([1, 1.6], [3, 4], [5, 6], capsize=2, fmt="none")
```
1.5.3

2.0

The error bars are not visible in 2.0.  Further inspection indicates that they are, in fact, drawn with the same color as the data line (that connects each data point), i.e., (0, 0, 0, 0).

Note that the docstring of errorbar is a bit inconsistent (emphasis mine):
> fmt : plot format string, optional, default: None
The plot format symbol. **If fmt is 'none' (case-insensitive),
only the errorbars are plotted.  This is used for adding
errorbars to a bar plot, for example.**
>ecolor : mpl color, optional, default: None
A matplotlib color arg which gives the color the errorbar lines;
**if None, use the color of the line connecting the markers.**

A strict reading of the doc for `ecolor` suggests that the current behavior is "correct", but the doc of `fmt` clearly indicates that this is not the intent.

### PR DESCRIPTION
**Pull request by dstansby**
Fixes #8117.

### CODE REVIEW

File: lib/matplotlib/tests/test_axes.py [hunk 1]
```diff
@@ -2454,6 +2454,17 @@ def test_errorbar_limits():
...
 
+def test_errobar_nonefmt():
+    # Check that passing 'none' as a format still plots errorbars
+    x = np.arange(5)
+    y = np.arange(5)
+
+    plotline, _, barlines = plt.errorbar(x, y, xerr=1, yerr=1, fmt='none')
+    assert plotline is None
+    for errbar in barlines:
+        assert np.any(errbar.get_color() != [0, 0, 0, 0])
```
Comment (dstansby) [inline lib/matplotlib/tests/test_axes.py]: I'm not sure if this is the best way to check that the errorbars aren't see-through, so other suggestions welcome.

Comment (tacaswell) [inline lib/matplotlib/tests/test_axes.py]: This color should be `'C0'`?  I would test against that.

File: lib/matplotlib/tests/test_axes.py [hunk 2]
```diff
@@ -2454,6 +2454,17 @@ def test_errorbar_limits():
     ax.set_title('Errorbar upper and lower limits')
 
 
+def test_errobar_nonefmt():
```
Comment (tacaswell) [inline lib/matplotlib/tests/test_axes.py]: still need `@cleanup` on 2.0.x.

**Comment by NelleV** [comment]
Anyone wants to give another look at this PR?

**Comment by afvincent** [comment]
Might be worth to re-run AppVeyor (which does not seem to have been run) before merging?

**Comment by tacaswell** [comment]
Appveyor does not run on the 2.0.x branch

**Comment by afvincent** [comment]
Ah, this might be the explanation then 😄!

**Pull request merged**